src/hevm/src/EVM/SymExec.hs

```diff
@@ -430,11 +430,8 @@ simplify e = if (mapExpr go e == e)
       | x == 0 = a
       | otherwise = o
 
-    -- Non-bitwise AND & OR -- no need to preserve bitwise equivalence.
-    go (ITE (And (Lit x) a) t f)
-      | x == 0 = f
-      | otherwise = ITE a t f
-    go (ITE (And a b@(Lit _)) t f) = ITE (And b a) t f
+    -- Non-bitwise OR -- no need to preserve bitwise equivalence.
```

Comment:
I actually find the comment here a little confusing, the Or is still bitwise. Maybe smth like this?
```suggestion
    -- If x is ever non zero the Or will always evaluate to some non zero value and the false branch will be unreachable
```